### DISCUSSION TITLE
AutoHDR v1.8.1 and later causes high CPU usage

### DISCUSSION THREAD

**Discussion by CaveWaverider** [Codectory/AutoActions#78] [Q&A]
Thanks for creating this wonderful tool AutoHDR!

However, it seems that there is a performance problem with version 1.8.1 and later. 

On my system, any version of AutoHDR (x64) from 1.8.1 - 1.8.7 is taxing the CPU heavily (~14-30% usage even when running in the background), causing the whole system to perform poorly and games to have low framerates and stutter. 

The last good version is 1.8.0 (aka 1.7.21), which taxes the CPU 0% and allows the system to perform well and games to run without stuttering (I can confirm 1.7.19 works fine, too).

System Specs:
CPU: Intel Core i9 9900K 8 Core/16 Threads
RAM: 64GB DDR4-3600 RAM
Video Card: Nvidia GeForce RTX 3090 24GB
Monitor: LG OLED 48CX, 3840x2160 Resolution, 150% Scaling
Operating System: Windows 11 Education 21H2 (Build 22000.376) 64-bit on SSD

Is anybody else experiencing the same issue?

**Comment by Codectory** [comment]
Is really 1.8.7 doing this? This was related to nvapi but I am using only necessary methods right now. So it's the same like 1.8.0

CaveWaverider ***@***.***> schrieb am So. 26. Dez. 2021 um
03:32:

**Reply by CaveWaverider** [reply]
Yes, I tested 1.8.7 briefly last night, it had the same CPU load issues. I must have installed 1.8.7 over 1.8.4 (I think) last night as I wanted to keep my applications, which saw CPU usage of 15-30%. 

However, I just tested 1.8.7 again, more thoroughly this time.

This time I installed it over 1.7.19 (which kept all the applications added with 1.7.19), and the CPU performance is at 0-0.3% (which is slightly worse than 1.8.0/1.7.19, but not bad and doesn't seem to affect CPU performance).

Afterwards I also tried to clean installed 1.8.7 and it also shows 0-0.3% CPU usage, so it's alright.

I assume some setting must have carried over from a bad version last night.

Sorry again about the report! Thanks again and keep up the good work, your tool really works wonders.
